services_app/src/main/java/org/opendatakit/services/MainActivity.java

```diff
@@ -143,38 +336,60 @@ public boolean onOptionsItemSelected(MenuItem item) {
       return true;
     }
 
-    if (id == R.id.action_resolve_conflict) {
+    if (id == R.id.action_settings) {
+
+      Intent intent = new Intent(this, AppPropertiesActivity.class);
+      intent.putExtra(IntentConsts.INTENT_KEY_APP_NAME, getAppName());
+      startActivityForResult(intent, SETTINGS_ACTIVITY_RESULT_CODE);
+      return true;
+    }
+
+    return false;
+  }
+
+  // Action on Clicking the Drawer Menu Item
+  private boolean onDrawerMenuItemClicked(MenuItem item){
+
+    drawerLayout.closeDrawer(GravityCompat.START);
+
+    if(item.getItemId()==R.id.drawer_resolve_conflict){
       Intent i = new Intent(this, AllConflictsResolutionActivity.class);
       i.putExtra(IntentConsts.INTENT_KEY_APP_NAME, getAppName());
       startActivityForResult(i, RESOLVE_CONFLICT_ACTIVITY_RESULT_CODE);
       return true;
     }
 
-    if (id == R.id.action_about) {
-
+    if(item.getItemId()==R.id.drawer_about_us){
       FragmentManager mgr = getSupportFragmentManager();
       GoToAboutFragment.GotoAboutFragment(mgr,R.id.main_activity_view);
       item.setVisible(false);
       return true;
     }
 
-    if (id == R.id.action_settings) {
-
+    if(item.getItemId()==R.id.drawer_settings){
       Intent intent = new Intent(this, AppPropertiesActivity.class);
       intent.putExtra(IntentConsts.INTENT_KEY_APP_NAME, getAppName());
       startActivityForResult(intent, SETTINGS_ACTIVITY_RESULT_CODE);
```

Comment:
I can see there are lot of intent related stuff happening, what do you think about having one utility classes having all these methods over there and you just pass configuration(activity class) and let that handle all intents for your application.